src/main/kotlin/interactor/GetAvgFruitNVeggieInteractor.kt

```diff
@@ -0,0 +1,49 @@
+package interactor
+
+import model.CityEntity
+import model.FruitAndVegetablesPrices
+class GetAvgFruitNVeggieInteractor(
+    private val dataSource: CostOfLivingDataSource,
+) {
+    fun execute(): List<CityEntity>? {
+        val list = dataSource.getAllCitiesData()
+        if (list.isEmpty()) return null
+        return dataSource.getAllCitiesData()
```

Comment:
Why are you calling getAllCitiesData again? you already have a reference to it `val list = dataSource.getAllCitiesData()`
```suggestion
        return list
```